core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -118,6 +119,7 @@ protected String tableType() {
         boolean ignoreResiduals, boolean caseSensitive, boolean colStats) {
       String schemaString = SchemaParser.toJson(schema());
       String specString = PartitionSpecParser.toJson(PartitionSpec.unpartitioned());
+      Map<Integer, PartitionSpec> specs = table().specs();
```

Comment:
Similar to the `ManifestsTable`, I think this needs to copy the specs into a `newHashMap`.

Reply:
updated